docs/plugins/hook.rst

```diff
@@ -0,0 +1,77 @@
+Hook Plugin
+===============
+
+Internally, beets sends events to plugins when an action finishes. These can
+range from importing a song (``import``) to beets exiting (``cli_exit``), and
+provide a very flexible way to perform actions based on the events. This plugin
+allows you to run commands when an event is emitted by beets, such as syncing
+your library with another drive when the library is updated.
+
+Hooks are currently run in the order defined in the configuration, however this
+is dependent on beets itself (may change in the future) and cannot be controlled
+by this plugin.
+
+.. _hook-configuration:
+
+Configuration
+-------------
+
+To configure the plugin, make a ``hook:`` section in your configuration
+file. The available options are:
+
+- **hooks**: A list of events and the commands to run (see :ref:`individual-hook-configuration`)
+  Default: Empty.
+- **substitute_event**: The string to replace in each command with the name of
+  the event executing it. This can be used to allow one script to act
+  differently depending on the event it was called by. Can be individually
+  overridden (see :ref:`individual-hook-configuration`).
+  Default: ``%EVENT%``
+- **shell**: Run each command in a shell. Can be individually overridden (see
+  :ref:`individual-hook-configuration`).
+  Default: ``yes``
```

Comment:
I've gone ahead and added the exception modification code in 8b4f349. It's not exactly pretty but it works. The error now looks like this:

```
Traceback (most recent call last):
  File "./beet", line 23, in <module>
    beets.ui.main()
  File "/Users/jack/Documents/Development/Personal/beets/beets/ui/__init__.py", line 1238, in main
    _raw_main(args)
  File "/Users/jack/Documents/Development/Personal/beets/beets/ui/__init__.py", line 1224, in _raw_main
    subcommands, plugins, lib = _setup(options, lib)
  File "/Users/jack/Documents/Development/Personal/beets/beets/ui/__init__.py", line 1100, in _setup
    plugins = _load_plugins(config)
  File "/Users/jack/Documents/Development/Personal/beets/beets/ui/__init__.py", line 1086, in _load_plugins
    plugins.send("pluginload")
  File "/Users/jack/Documents/Development/Personal/beets/beets/plugins.py", line 458, in send
    result = handler(**arguments)
  File "/Users/jack/Documents/Development/Personal/beets/beets/plugins.py", line 123, in wrapper
    return func(*args, **kwargs)
  File "/Users/jack/Documents/Development/Personal/beets/beetsplug/hook.py", line 61, in hook_function
    subprocess.Popen(command_pieces).wait()
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory: not_a_real_command
```

I admit it could still do with some work though, as it's not inherently obvious that it's from the hook plugin.